### PR TITLE
Fix to handle weird scenarios where the request urls to assets contain query string

### PR DESCRIPTION
**Pull request by infindex**

### CODE REVIEW

**Comment by jrgm** [comment]
So, for one thing, the tests no longer pass. Also a description of the problem you are trying to solve would help.

File: lib/connect-cachify.js [hunk 1]
```diff
@@ -9,7 +9,8 @@ var crypto = require('crypto'),
     format = require('./compat').format,
     fs = require('fs'),
     path = require('path'),
-    und = require('underscore');
+    und = require('underscore'),
+    URL = require('url');
```
Comment (ozten) [inline lib/connect-cachify.js]: variable holding the module should be lowercase.

File: lib/connect-cachify.js [hunk 2]
```diff
@@ -114,7 +117,7 @@ exports.setup = function (assets, options) {
 
         if (exists === true && hash === actualHash) {
           resp.setHeader('Cache-Control', 'public, max-age=31536000');
-          req.url = url;
+          req.url = url + '?' + req_url.query; // append back the query string
```
Comment (ozten) [inline lib/connect-cachify.js]: Awesome! Can you add a test that demonstrates the problem that query string parameters introduce?

**Comment by infindex** [comment]
@jrgm  - fixed the test failures...
@ozten - additional test cases added to demonstrate the query string problem. also fixed the variable name as suggested.

Also, added query string parameter handling in development mode and urls with Hash ids.

Thanks!
MG

File: test/connect-cachify-test.js [hunk 3]
```diff
@@ -261,6 +334,32 @@ exports.setup = nodeunit.testCase({
...
+      resp = get_resp(),
+      mddlwr;
+
+    mddlwr = cachify.setup(
+      assets, {
+        prefix: 'cachify',
+        root: '/tmp'
+      });
+    var link = cachify.cachify_js("/other", {hash: 'd41d8cd98f'});
+    test.equal(link, '<script src="/cachify/d41d8cd98f/other"></script>');
```
Comment (ozten) [inline test/connect-cachify-test.js:351]: Awesome, thanks for adding all these tests.

**Pull request merged**